artiq/devices/novatech409B/novatech409B.py

```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2015 Joe Britton
+
+import serial
+import sys
+import time
+import math
+import platform
+import logging
+import inspect
+
+
+class Novatech409B:
+    """controller for Novatech 409B 4-channel DDS
+
+    This class is an interface with the Novatech Model 409B 4-
+    channel DDS box. The interface is a serial interface.
+
+    :param int comport: COM port number on Windows
+    :param int debug: debug level
+    """
+    # NOTE: the in-line documentation is in Sphinx format
+    # http://sphinx-doc.org/domains.html
+    def __init__(self, comport=1, debug=1, simulate_hw=False):
+        # some private members
+        self.__comport = comport
+        self.__debug = debug
+        self.__className = "Novatech409B"
+        self.simulate_hw = simulate_hw  # true if disconnected from hw
+        self.serial_rw_delay = 0.001  # is time between reads and writes
+
+        # setup logging
+        FORMAT = "%(asctime)-15s %(message)s"
+        logging.basicConfig(format=FORMAT)
+        self.logger = logging.getLogger("artiq.driver.novatech409B")
+        self.logger.setLevel(logging.DEBUG)
+        self.debug_message("__init__", "", level=3)
+
+        # establish serial connection --- platform dependent
+        if not self.simulate_hw:
+            self.__platform = platform.system()
+            if self.__platform == "Windows":
+                # note that pySerial starts counting serial ports at 0
+                serial_port_id = int(self.__comport)-1
+            elif self.__platform is "Linux":
+                # just use the device string as
+                # passed (e.g. "/dev/ttyUSB0")
+                serial_port_id = "/dev/ttyUSB0"
+            else:
+                self.debug_message("__init__", "unknown platform", level=0)
+                sys.exit()
+            self.__ser = serial.Serial(
+                serial_port_id,
+                baudrate=19200,
+                bytesize=8,
+                parity="N",
+                stopbits=1,
+                xonxoff=0,
+                timeout=0.05)
+        self.setup()
+
+    def __del__(self):
```

Comment:
https://docs.python.org/3/reference/datamodel.html#object.__del__
You don't need `__del__` here. If the instance is collected, its attributes collected if they are not otherwise referenced. `__ser.close()` will be called anyway. What problem were you trying to solve?